jobbergate-api/jobbergate_api/tests/apps/applications/test_routers.py

```diff
@@ -262,23 +264,32 @@ async def test_delete_application__fk_error(
     count = await database.fetch_all("SELECT COUNT(*) FROM applications")
     assert count[0][0] == 1
 
-    inject_security_header("owner1@org.com", Permissions.APPLICATIONS_EDIT)
-    with mock.patch(
-        "jobbergate_api.storage.database.execute",
-        side_effect=asyncpg.exceptions.ForeignKeyViolationError(
-            f"""
-            update or delete on table "applications" violates foreign key constraint
-            "job_scripts_application_id_fkey" on table "job_scripts"
-            DETAIL:  Key (id)=({inserted_id}) is still referenced from table "job_scripts".
-            """
+    await database.execute(
+        query=job_scripts_table.insert(),
+        values=dict(
+            **job_script_data,
+            application_id=inserted_id,
         ),
-    ):
-        response = await client.delete(f"/jobbergate/applications/{inserted_id}")
-    assert response.status_code == status.HTTP_409_CONFLICT
-    error_data = json.loads(response.text)["detail"]
-    assert error_data["message"] == "Delete failed due to foreign-key constraint"
-    assert error_data["table"] == "job_scripts"
-    assert error_data["pk_id"] == f"{inserted_id}"
+    )
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_scripts")
+    assert count[0][0] == 1
```

Comment:
Assert if there's only one job script in the database doesn't guarantee the job script is the correct one. There should be assertions against each row of the job script IMO

Reply:
I agree with you. However, this mechanism is used throughout the Jobbergate API code base. If we change it in one place, we should change it throughout. However, that's some overhead I don't want to take on right now. So, I just followed the already established pattern in the new tests.